### ISSUE TITLE
1.8.1: 'vagrant up' doesn't support regexp anymore

### ISSUE THREAD

**Issue by taurus-forever** [mitchellh/vagrant#6828]
Hi,

Previously (vagrant 1.7.4 at least) it was possible to start several machines using one command with regexp, like:
```
vagrant up /test*/
```
(assuming I have the following vagrant machines available: test1 test2 test3 )

Can you please confirm it is expected result for Vagrant 1.8.1 now:
```
> vagrant up /test*/
...
The machine with the name '/test*/' was not found configured for
this Vagrant environment.
```

I do not see the following part of code in vagrant 1.8.1 debug output anymore:
```
> VAGRANT_LOG=debug vagrant up /test*/
...
DEBUG command: Getting target VMs for command. Arguments:
DEBUG command:  -- names: ["/test*/"]
DEBUG command:  -- options: {:provider=>nil}
DEBUG command: Finding machines that match regex: test*
...

Thank you!
```

**Comment by sethvargo** [comment]
Hi @tarsus-forever,

I just tried Vagrant 1.8.1 (and master), and the regular expression pattern matching is still working for me:

```ruby
Vagrant.configure(2) do |config|
  config.vm.box = "hashicorp/precise64"

  config.vm.provider "virtualbox" do |vb|
   vb.cpus = 1
   vb.memory = 1024
  end

  config.vm.define "web-1"
  config.vm.define "web-2"
  config.vm.define "web-3"
end
```

```
$ vagrant up '/web-*/'
Bringing machine 'web-1' up with 'virtualbox' provider...
Bringing machine 'web-2' up with 'virtualbox' provider...
Bringing machine 'web-3' up with 'virtualbox' provider...
==> web-1: Importing base box 'hashicorp/precise64'...
```

Note that I had to add single quotes to prevent ZSH from escaping the argument before passing to Vagrant.

Can you please share any additional information to help us reproduce this issue

**Comment by taurus-forever** [comment]
Hm, tested your version, the same state:
```
23:07:23 ✔ taurus:~$ mkdir -p /tmp/test
23:07:30 ✔ taurus:~$ cd /tmp/test/
23:07:33 ✔ taurus:/tmp/test$ cat > Vagrantfile << EOF
> Vagrant.configure(2) do |config|
>   config.vm.box = "hashicorp/precise64"
> 
>   config.vm.provider "virtualbox" do |vb|
>    vb.cpus = 1
>    vb.memory = 1024
>   end
> 
>   config.vm.define "web-1"
>   config.vm.define "web-2"
>   config.vm.define "web-3"
> end
> EOF
23:07:58 ✔ taurus:/tmp/test$ vagrant up '/web-*/'
The machine with the name '/web-*/' was not found configured for
this Vagrant environment.
23:08:09 ✘ taurus:/tmp/test$ vagrant up 'web-1'
Bringing machine 'web-1' up with 'virtualbox' provider...
==> web-1: Box 'hashicorp/precise64' could not be found. Attempting to find and install...
    web-1: Box Provider: virtualbox
    web-1: Box Version: >= 0
==> web-1: Loading metadata for box 'hashicorp/precise64'
    web-1: URL: https://atlas.hashicorp.com/hashicorp/precise64
==> web-1: Adding box 'hashicorp/precise64' (v1.1.0) for provider: virtualbox
    web-1: Downloading: https://atlas.hashicorp.com/hashicorp/boxes/precise64/versions/1.1.0/providers/virtualbox.box
    web-1: Progress: 100% (Rate: 59/s, Estimated time remaining: --:--:--)
...
23:16:40 ✔ taurus:/tmp/test$ vagrant status
Current machine states:

web-1                     not created (virtualbox)
web-2                     not created (virtualbox)
web-3                     not created (virtualbox)

This environment represents multiple VMs. The VMs are all listed
above with their current state. For more information about a specific
VM, run `vagrant status NAME`.
23:16:44 ✔ taurus:/tmp/test$ 
```
We probably need to compare environment:
 - Linux Mint 17.3 with the latest available packages
 - Vagrant 1.8.1 from https://releases.hashicorp.com/vagrant/1.8.1/vagrant_1.8.1_x86_64.deb
 - Ruby related packages installed:
```
23:14:44 ✔ taurus:~$ dpkg -l | grep ii | grep -i ruby
ii  libruby1.9.1                                                1.9.3.484-2ubuntu1.2                                amd64        Libraries necessary to run Ruby 1.9.1
ii  libruby2.0:amd64                                            2.0.0.484-1ubuntu2.2                                amd64        Libraries necessary to run Ruby 2.0
ii  ruby                                                        1:1.9.3.4                                           all          Interpreter of object-oriented scripting language Ruby (default version)
ii  ruby-net-ssh                                                1:2.6.8-1                                           all          Ruby implementation of the SSH protocol
ii  ruby1.9.1                                                   1.9.3.484-2ubuntu1.2                                amd64        Interpreter of object-oriented scripting language Ruby
ii  ruby2.0                                                     2.0.0.484-1ubuntu2.2                                amd64        Interpreter of object-oriented scripting language Ruby
ii  rubygems-integration                                        1.5                                                 all          integration of Debian Ruby packages with Rubygems
23:14:46 ✔ taurus:~$ 
```

**Comment by sethvargo** [comment]
Hi @taurus-forever 

I am running OS X, so I'm using a different OS and package, but the same version. The rubies installed on the system do not matter because Vagrant bundles it's own Ruby that is completely separate from any system libs.

What shell are you using?

**Comment by taurus-forever** [comment]
It is a bash which is coming out of the box:
$ apt-cache policy bash
bash:
  Installed: 4.3-7ubuntu1.5
  Candidate: 4.3-7ubuntu1.5

The same shell works well for Vagrant 1.7.4 :-)

**Comment by taurus-forever** [comment]
Here the relative part of debug from 1.7.4 (works well):
```
$ VAGRANT_LOG=debug vagrant up '/web-*/'
...
 INFO runner: Running action: environment_load #<Vagrant::Action::Builder:0x0000000350c8c0>
 INFO cli: CLI: [] "up" ["/web-*/"]
DEBUG cli: Invoking command class: VagrantPlugins::CommandUp::Command ["/web-*/"]
DEBUG command: 'Up' each target VM...
DEBUG command: Getting target VMs for command. Arguments:
DEBUG command:  -- names: ["/web-*/"]
DEBUG command:  -- options: {:provider=>nil}
DEBUG command: Finding machines that match regex: web-*
 INFO loader: Set :root = #<Pathname:/tmp/test/Vagrantfile>
DEBUG loader: Populating proc cache for #<Pathname:/tmp/test/Vagrantfile>
DEBUG loader: Load procs for pathname: /tmp/test/Vagrantfile
 INFO loader: Loading configuration in order: [:home, :root]
DEBUG loader: Loading from: root (evaluating)
DEBUG loader: Configuration loaded successfully, finalizing and returning
DEBUG push: finalizing
```

Vagrant 1.8.1:
```
$ VAGRANT_LOG=debug vagrant up '/web-*/'
...
 INFO runner: Running action: environment_load #<Vagrant::Action::Builder:0x000000024298f0>
 INFO cli: CLI: [] "up" ["/web-*/"]
DEBUG cli: Invoking command class: VagrantPlugins::CommandUp::Command ["/web-*/"]
DEBUG command: 'Up' each target VM...
 INFO loader: Set :root = ["#<Pathname:/tmp/test/Vagrantfile>"]
DEBUG loader: Populating proc cache for #<Pathname:/tmp/test/Vagrantfile>
DEBUG loader: Load procs for pathname: /tmp/test/Vagrantfile
 INFO loader: Loading configuration in order: [:home, :root]
DEBUG loader: Loading from: root (evaluating)
DEBUG loader: Configuration loaded successfully, finalizing and returning
DEBUG push: finalizing
...
```

**Comment by sethvargo** [comment]
I'm going to tag this as a bug, but I'm still unable to reproduce. It might be something that changed on linux.

**Comment by taurus-forever** [comment]
One more comment here after checking diff of debug Vagrant 1.7.4 and 1.8.1.
1.7.4 had loaded the following plugin:
```
DEBUG global: Loading core plugin: /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/plugins/guests/debian8/plugin.rb
 INFO manager: Registered plugin: Debian Jessie guest
```
while 1.8.1 has no such plugin anymore. 

I can be related to https://github.com/apertoso/vagrant/commit/e0dad41b0c04de7a35b4fbf44b23eea1b41d1e4e and probably somehow affects my system as I am using Debian 8 on both host machine and guest VM.

There are no more difference in diff of debug Vagrant 1.7.4 and 1.8.1.

**Comment by garrettr** [comment]
I'm also having this problem on Mac OS 10.10.5.

**Expected behavior**

`vagrant status` shows that are two machines in my environment whose names ends in `staging`, `app-staging` and `mon-staging`. I expect that running `vagrant up /staging$/` will start both machines.

**Actual behavior**

`vagrant up /staging$/` returns the following error:

```
$ vagrant up /staging$/                  
The machine with the name '/staging$/' was not found configured for
this Vagrant environment.
```

[Gist of output with `VAGRANT_LOG=debug`](https://gist.github.com/garrettr/355799fcfe9b0bd27381).

**Comment by ddavisgso** [comment]
Just in case it helps, on windows here... 

c:\projects\vagrant\vagrant>vagrant status /^loghost/
Current machine states:

loghost                   poweroff (virtualbox)
loghostb                  poweroff (virtualbox)

This environment represents multiple VMs. The VMs are all listed
above with their current state. For more information about a specific
VM, run `vagrant status NAME`.

c:\projects\vagrant\vagrant>vagrant up /^loghost/
The machine with the name '/loghost/' was not found configured for
this Vagrant environment.

c:\projects\vagrant\vagrant>vagrant -v
Vagrant 1.8.1

**Comment by taurus-forever** [comment]
Resolved as duplicate of https://github.com/mitchellh/vagrant/issues/6908
Please consider to include the fix in Vagrant 1.8.2
Thank you for the fix @willonymous !

**Comment by garrettr** [comment]
@taurus-forever Do you (or anybody else from Vagrant) know when 1.8.2 will be released? It's been more than 3 months since the fix was committed and this is still an issue.
